src/components/AppFooter.tsx

```diff
@@ -0,0 +1,136 @@
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import styled, { css } from 'styled-components'
+
+import { ReactComponent as DiscordIcon } from '../images/brand-icon-discord.svg'
+import { ReactComponent as RedditIcon } from '../images/brand-icon-reddit.svg'
+import { ReactComponent as TelegramIcon } from '../images/brand-icon-telegram.svg'
+import { ReactComponent as TwitterIcon } from '../images/brand-icon-twitter.svg'
+import { deviceBreakPoints } from '../style/globalStyles'
+import ExternalLink from './ExternalLink'
+import NetworkSwitch from './NetworkSwitch'
+import ThemeSwitcher from './ThemeSwitcher'
+
+interface AppFooterProps {
+  className?: string
+}
+
+const AppFooter = ({ className }: AppFooterProps) => (
+  <footer className={className}>
+    <LeftGroup>
+      <StyledNetworkSwitch direction="up" />
+      <ThemeSwitcher />
+    </LeftGroup>
+    <RightGroup>
+      <SocialMediaIconList>
+        {socialMediaData.map((d) => (
+          <ExternalLink href={d.link} key={d.name}>
+            <d.Icon data-tip={d.name} className="social-media-icon" />
+          </ExternalLink>
+        ))}
+      </SocialMediaIconList>
+      <span>
+        <ExternalLink href="https://github.com/alephium/explorer">Source code ↗</ExternalLink>
+      </span>
+      <ExternalLink href="https://alephium.org">Alephium.org ↗</ExternalLink>
+      <span>Powered by Alephium - {new Date().getFullYear()}</span>
+    </RightGroup>
+  </footer>
+)
+
+const socialMediaData = [
+  {
+    name: 'Discord',
+    link: 'https://discord.gg/JErgRBfRSB',
+    Icon: DiscordIcon
+  },
+  {
+    name: 'Telegram',
+    link: 'https://t.me/alephiumgroup',
+    Icon: TelegramIcon
+  },
+  {
+    name: 'Reddit',
+    link: 'https://www.reddit.com/r/Alephium',
+    Icon: RedditIcon
+  },
+  {
+    name: 'Twitter',
+    link: 'https://twitter.com/alephium',
+    Icon: TwitterIcon
+  }
+]
+
+export default styled(AppFooter)`
+  display: flex;
+  justify-content: space-between;
+  background-color: ${({ theme }) => theme.bgTertiary};
+  padding: 15px 30px;
+  border-top: 1px solid ${({ theme }) => theme.borderSecondary};
+
+  @media ${deviceBreakPoints.mobile} {
+    flex-direction: column;
+    gap: 30px;
+  }
+`
+
+const FooterGroup = css`
+  display: flex;
+  align-items: center;
+  gap: 20px;
+
+  @media ${deviceBreakPoints.mobile} {
+    flex-direction: column;
+  }
+`
+
+const LeftGroup = styled.div`
+  ${FooterGroup}
+  justify-self: flex-start;
+`
+
+const RightGroup = styled.div`
+  ${FooterGroup}
```

Comment:
Instead of this, what do you think about the following?

```js
const FooterGroup = styled.div`
  display: flex;
  align-items: center;
  gap: 20px;

  @media ${deviceBreakPoints.mobile} {
    flex-direction: column;
  }
`

const LeftGroup = styled(FooterGroup)`
  justify-self: flex-start;
`

const RightGroup = styled(FooterGroup)`
   ...
```

This way, styled components will generate less CSS code, since it will generate a class for `FooterGroup` that has the shared code and assign it to both HTML divs (`LeftGroup` and `RightGroup`):

```css
.jAvptK {
    justify-self: flex-end;
}

.ldwtwo {
    justify-self: flex-start;
}

.eGQqvL {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 20px;
}
```

Currently, this is the generated CSS code:

```css
.ktCfcD {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 20px;
    justify-self: flex-start;
}

.dRZtmo {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 20px;
    justify-self: flex-end;
}
```

Reply:
Looks good! Done.